### PR TITLE
Char literal for amino acids, removal of enum interface and introduction of is_aminoacid metafunction

### LINKED ISSUE #641: **amino acid alphabets**: char-literal instead of enum-notation

### PR DESCRIPTION
**Pull request by smehringer**
Resolves #641

### CODE REVIEW

File: include/seqan3/alphabet/aminoacid/aa20.hpp [hunk 1]
```diff
@@ -184,13 +184,29 @@ using aa20_vector = std::vector<aa20>;
...
 {
 
-/*!\brief aa20 literal
+/*!\name Literals
+ * \{
+ */
+
+/*!\brief The seqan3::aa20 char literal.
+ * \relates seqan3::aa20
+ * \returns seqan3::aa20
```
Comment (joergi-w) [inline include/seqan3/alphabet/aminoacid/aa20.hpp]: `\param[in] c` missing

File: include/seqan3/alphabet/aminoacid/aa27.hpp [hunk 2]
```diff
@@ -167,7 +167,23 @@ using aa27_vector = std::vector<aa27>;
...
 {
 
-/*!\brief aa27 literal
+/*!\name Literals
+ * \{
+ */
+
+/*!\brief The seqan3::aa27 char literal.
+ * \relates seqan3::aa27
+ * \returns seqan3::aa27
```
Comment (joergi-w) [inline include/seqan3/alphabet/aminoacid/aa27.hpp]: also here

File: include/seqan3/alphabet/aminoacid/aa20.hpp [hunk 3]
```diff
@@ -184,13 +184,29 @@ using aa20_vector = std::vector<aa20>;
...
+ *
+ * \snippet test/snippet/alphabet/aminoacid/aa20.cpp char_literal
+ *
+ */
+constexpr aa20 operator""_aa20(char const c) noexcept
+{
+    return aa20{}.assign_char(c);
+}
+
+/*!\brief The seqan3::aa20  string literal
```
Comment (h-2) [inline include/seqan3/alphabet/aminoacid/aa20.hpp]: . missing at end of line

File: test/unit/alignment/scoring_scheme_test.cpp [hunk 4]
```diff
@@ -291,10 +291,10 @@ TYPED_TEST(generic, convertability)
         {
             using nucl_t = std::decay_t<decltype(aa)>;
 
-            EXPECT_EQ(scheme.score(aa27::C,                   aa27::G),
+            EXPECT_EQ(scheme.score('C'_aa27,                   'G'_aa27),
                       scheme.score(nucl_t{}.assign_char('C'), nucl_t{}.assign_char('G')));
-            EXPECT_EQ(scheme.score(aa27::T,                   nucl_t{}.assign_char('T')),
-                      scheme.score(nucl_t{}.assign_char('T'), aa27::T));
+            EXPECT_EQ(scheme.score('T'_aa27,                   nucl_t{}.assign_char('T')),
```
Comment (joergi-w) [inline test/unit/alignment/scoring_scheme_test.cpp]: check indentation

File: test/unit/alphabet/composition/composition_test.cpp [hunk 5]
```diff
@@ -269,10 +269,10 @@ TEST(composition, custom_comparison)
     EXPECT_EQ('C'_rna4,    t11);
     EXPECT_EQ(phred42{3}, t11);
 
-    qualified<aa27, phred63> t21{aa27::K, phred63{3}};
-    EXPECT_EQ(t21, aa27::K);
+    qualified<aa27, phred63> t21{'K'_aa27, phred63{3}};
+    EXPECT_EQ(t21, 'K'_aa27);
     EXPECT_EQ(t21, phred63{3});
-    EXPECT_EQ(aa27::K,     t21);
+    EXPECT_EQ('K'_aa27,     t21);
```
Comment (joergi-w) [inline test/unit/alphabet/composition/composition_test.cpp]: check indentation

**Comment by h-2** [comment]
CI fails...

File: include/seqan3/alphabet/aminoacid/concept.hpp [hunk 6]
```diff
@@ -48,54 +48,32 @@
 namespace seqan3
 {
 
+//!\brief Helper metafunction that identifies amino acid alphabets.
+//!\ingroup aminoacid
```
Comment (h-2) [inline include/seqan3/alphabet/aminoacid/concept.hpp]: I would add something like this to \details

### Customisation point

If you define your own alphabet and want it to be recognised as an amino acid alphabet by SeqAn, you need to specialise this metafunction for your type and have it inherit std::true_type. 

```cpp
namespace your_namespace
{

struct your_aa
{
//...
};

}

namespace seqan3
{

template <>
struct is_aminoacid<your_namespace::your_aa> : std::true_type {};

}
```

File: include/seqan3/alphabet/aminoacid/concept.hpp [hunk 7]
```diff
@@ -48,54 +48,47 @@
...
+ *
+ * Since an amino acid alphabet has no specific characteristics (like the complement
+ * function for nucleotide alphabets), we distinguish an amino acid alphabet by
+ * the seqan3::is_aminoacid metafunction.
+ *
+ * ### Customisation point
+ *
+ * If you define your own alphabet and want it to be recognised as an amino acid
+ * alphabet by SeqAn, you need to specialise this metafunction for your type and
+ *  have it inherit std::true_type.
```
Comment (h-2) [inline include/seqan3/alphabet/aminoacid/concept.hpp]: double space :nail_care:

**Pull request merged**